config/network.go

```diff
@@ -0,0 +1,161 @@
+	if err != nil {
+		return fmt.Errorf("error unmarshaling network config: %w", err)
+	}
+
+	err = n.ApplyOverrides(&cfg)
+	if err != nil {
+		return fmt.Errorf("error applying overrides from decoded network config file to config: %w", err)
+	}
+
+	return nil
+}
+
+func (n *NetworkConfig) ApplyBase64Enconded(configEncoded string) error {
+	if configEncoded == "" {
+		return nil
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(configEncoded)
+	if err != nil {
+		return err
+	}
+
+	return n.ApplyDecoded(string(decoded))
+}
+
+func (n *NetworkConfig) Validate() error {
+	if len(n.SelectedNetworks) == 0 {
+		return errors.New("selected_networks must be set")
+	}
+
+	upperCasedHttpUrls := make(map[string][]string)
+	for key := range n.RpcHttpUrls {
+		upperCasedHttpUrls[strings.ToUpper(key)] = n.RpcHttpUrls[key]
+	}
+	n.RpcHttpUrls = upperCasedHttpUrls
+
+	upperCasedWsUrls := make(map[string][]string)
+	for key := range n.RpcWsUrls {
+		upperCasedWsUrls[strings.ToUpper(key)] = n.RpcWsUrls[key]
+	}
+	n.RpcWsUrls = upperCasedWsUrls
+
+	upperCasedWalletKeys := make(map[string][]string)
+	for key := range n.WalletKeys {
+		upperCasedWalletKeys[strings.ToUpper(key)] = n.WalletKeys[key]
+	}
+	n.WalletKeys = upperCasedWalletKeys
+
+	for i, network := range n.SelectedNetworks {
+		n.SelectedNetworks[i] = strings.ToUpper(network)
+	}
+
+	for _, network := range n.SelectedNetworks {
+		if strings.Contains(network, "SIMULATED") {
+			// we don't need to validate RPC endpoints or private keys for simulated networks
+			continue
+		}
+
+		if _, ok := n.RpcHttpUrls[network]; !ok {
+			return fmt.Errorf("At least one HTTP RPC endpoint for %s network must be set", network)
+		}
+
+		if _, ok := n.RpcWsUrls[network]; !ok {
+			return fmt.Errorf("At least one WS RPC endpoint for %s network must be set", network)
+		}
+
+		if _, ok := n.WalletKeys[network]; !ok {
+			return fmt.Errorf("At least one private key of funding wallet for %s network must be set", network)
+		}
+	}
+
+	return nil
+}
+
+func (n *NetworkConfig) ApplyOverrides(from *NetworkConfig) error {
+	if from == nil {
+		return nil
+	}
+	if from.SelectedNetworks != nil {
+		n.SelectedNetworks = from.SelectedNetworks
+	}
+	if from.RpcHttpUrls != nil {
+		if n.RpcHttpUrls == nil || len(n.RpcHttpUrls) == 0 {
+			n.RpcHttpUrls = from.RpcHttpUrls
+		} else {
+			for network, urls := range from.RpcHttpUrls {
+				n.RpcHttpUrls[network] = urls
+			}
+		}
+	}
+	if from.RpcWsUrls != nil {
+		if n.RpcWsUrls == nil || len(n.RpcWsUrls) == 0 {
+			n.RpcWsUrls = from.RpcWsUrls
+		} else {
+			for network, urls := range from.RpcWsUrls {
+				n.RpcWsUrls[network] = urls
+			}
+		}
+	}
+	if from.WalletKeys != nil {
+		if n.WalletKeys == nil || len(n.WalletKeys) == 0 {
+			n.WalletKeys = from.WalletKeys
+		} else {
+			for network, urls := range from.WalletKeys {
+				n.WalletKeys[network] = urls
+			}
+		}
+	}
+
+	return nil
+}
+
+func (n *NetworkConfig) Default() error {
```

Comment:
Should we add default from env vars?

Reply:
Do you mean making the `Default()` actually execute [this method](https://github.com/smartcontractkit/chainlink-testing-framework/pull/802/files/85ab90ffd3b5456bad3f1477d60fcb07014d9067#diff-32f4f366dad3cee33f9165dbf93cf213721a7dd734bf3e99cd295e35ba122e4dR26)? Or something yet different?

Reply:
yes

Reply:
I modified it so that what `Defaults()` actually does is read env var